lib/std/fmt.zig

```diff
@@ -664,6 +664,10 @@ pub fn formatType(
             if (max_depth == 0) {
                 return writer.writeAll("{ ... }");
             }
+            if (info.sentinel) |s| {
+                const index = std.mem.indexOfSentinel(info.child, @as(*info.child, @constCast(@ptrCast(s))).*, value[0..]);
```

Comment:
`@constCast` changes a pointer-to-const `*const T` to a pointer-to-mutable `*T`.
For reading the value here, pointer-to-const should already be sufficient:
`const index = std.mem.indexOfSentinel(info.child, @as(*const info.child, @ptrCast(s)).*, value[0..]);`

Reply:
thanks